apps/full-site-editing/bin/babel-transform.js

```diff
@@ -0,0 +1,8 @@
+/**
```

Comment:
I don't know if this dir is the best place for this file, although it felt natural to put it at the same level as the file that is consuming it. I'm open to suggestions.

Reply:
By the way, I'd love some insights about something: FSE uses `calypso-build` which **already** provides some Jest presets. However, it's using presets from `@wordpress/scripts/config/jest-unit.config.js` instead. 

I did try using the presets from Calypso and while the (TS) suite in this changeset worked, I started having errors in other FSE tests, related to `import` being used outside a module context, `React` not being defined and `window` not found (the latest two were easy to fix, for undefined `window`, I just needed to set the `testEnvironment` jest config setting to `jsdom` and for the undefined `React`, I just needed to import `React`). It ended up being more trouble than it's worth, so I just decided to keep the ways things were and just append the `typescript-preset` here for the transformer here. This works well.

Reply:
With this package using `calypso-build` to build, I'd like to use its jest preset: `@automattic/calypso-build`

https://github.com/Automattic/wp-calypso/blob/master/packages/calypso-build/jest-preset.js

Reply:
That's the same reasoning I used and I tried using this preset, but while TS worked fine, I go additional errors:||

> I did try using the presets from Calypso and while the (TS) suite in this changeset worked, I started having errors in other FSE tests, related to import being used outside a module context, React not being defined and window not found (the latest two were easy to fix, for undefined window, I just needed to set the testEnvironment jest config setting to jsdom and for the undefined React, I just needed to import React).

I'll give it another try.

Reply:
I _very much dislike adding more customization and configuration here_. That's not a critique of this PR or your work, but a critique of the current state of this app's setup.

It has one foot in Calypso build (which has many examples of working _just fine_ with TypeScript in this repo). It has another foot in the `@wordpress/*` side of things (wp-scripts, etc.).

Most of the build seems to be handled by `calypso-build`, but most of the testing seems to be run via `wp-scripts` and its family of helpers. I'd love to see this package commit to one direction or the other, honestly I don't care too much, but committing to either of these tooling solutions would be a better solution.

As I see it right now, we have two alternatives, both of which should work well and remove the amount of configuration we need to include here. However, we've wound up in a situation where we need another complete set of configuration on top of two solutions in order to smooth over the incompatibilities.

😤

I'll set aside my frustrations now because we're not here to solve the setup issues in this PR. I know I started us down this road by suggestions TypeScript, but it's very troubling to me that it doesn't work.

We can ignore these concerns for this PR.

(👋  tooling partner in crime @ockham)